application/client.core/src/app/environment/services/service.file.opener.ts

```diff
@@ -162,6 +163,24 @@ export class FileOpenerService implements IService, IFileOpenerService {
         this._open(checked, EActionType.concat, session);
     }
 
+    public openEach(list: IPCMessages.IFile[] | FilesList) {
+        const checked: IPCMessages.IFile[] = list instanceof Array ? this._filterChecked(list) : this._filterChecked(list.getFiles());
+        if (checked.length <= 0) {
+            return;
+        }
+        const openFile = (index: number) => {
+            if (index >= checked.length) {
+                return;
+            }
+            this.openFileByName(checked[index].path).then(() => {
+                openFile(++index);
+            }).catch((error: Error) => {
+                this._logger.error(`Fail to open file ${checked[index].path} due error: ${error.message}`);
```

Comment:
I think we need to notify user about failing to open the file. And probably we still shoud try to open rest files.